src/data/apigateway/mod.rs

```diff
@@ -0,0 +1,170 @@
+#[cfg(test)]
+mod tests;
+
+pub mod auth;
+
+use chrono::prelude::*;
+use std::fmt;
+use std::collections::HashMap;
+use serde;
+use serde_qs as qs;
+
+#[derive(Debug,Eq,PartialEq,Serialize,Deserialize)]
+pub enum HttpMethod {
+    HEAD,
+    GET,
+    POST,
+    PUT,
+    OPTIONS,
+    DELETE,
+}
+
+impl fmt::Display for HttpMethod {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", match *self {
+            HttpMethod::HEAD => "HEAD",
+            HttpMethod::GET => "GET",
+            HttpMethod::POST => "POST",
+            HttpMethod::PUT => "PUT",
+            HttpMethod::OPTIONS => "OPTIONS",
+            HttpMethod::DELETE => "DELETE"
+        })
+    }
+}
+
+#[derive(Serialize,Deserialize)]
+pub struct HttpEvent {
+    pub body: Option<String>,
+    pub headers: Option<HashMap<String, String>>,
+    #[serde(rename="httpMethod")]
+    pub http_method: HttpMethod,
+    #[serde(rename="isBase64Encoded")]
+    pub is_base64_encoded: bool,
+    pub path: String,
+    #[serde(rename="pathParameters")]
+    pub path_parameters: Option<HashMap<String, String>>,
+    #[serde(rename="queryStringParameters")]
+    pub query_string_parameters: Option<HashMap<String, String>>,
+    pub resource: String,
+    #[serde(rename="requestContext")]
+    pub request_context: HttpEventRequestContext,
+    #[serde(rename="stageVariables")]
+    pub stage_variables: Option<HashMap<String, String>>,
+}
+
+impl HttpEvent {
+
+    pub fn get_header(&self, key: &str) -> Option<&str> {
```

Comment:
There are some some growing community recommendations on avoiding get_ where you can https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Instead a simple `.header("Foo")` will do

Reply:
Roger that.